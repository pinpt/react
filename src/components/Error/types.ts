import { ReactElement, ReactNode } from 'react';
import { IActionLinkProps } from '../Internal/ActionLink';
import { ILogoProps } from '../Logo';

export interface IErrorProps {
	className?: string;
	logo?: ReactElement<ILogoProps | IActionLinkProps>;
	error?: string;
	title?: string;
	description?: string;
	link?: ReactNode;
}
