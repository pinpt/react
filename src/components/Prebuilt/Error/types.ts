import { ReactElement } from "react";
import { ISite } from "../../../lib";

export interface IPrebuiltErrorProps {
	renderLogo?: (site) => ReactElement;
	site: ISite;
};
