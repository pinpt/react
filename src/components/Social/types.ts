export interface ISocialProps {
	className?: string;
	newTab?: boolean;
	href?: string;
	onClick?: () => void;
}
