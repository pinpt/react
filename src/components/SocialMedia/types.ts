export interface ISocialMediaLinkProps {
	className?: string;
	newTab?: boolean;
	href?: string;
	onClick?: () => void;
}

export interface ISocialMediaShareProps {
	className?: string;
	newTab?: boolean;
	url: string;
	onClick?: () => void;
}
