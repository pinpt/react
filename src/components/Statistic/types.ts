export interface IStatisticProps {
	count: number;
	className?: string;
}
