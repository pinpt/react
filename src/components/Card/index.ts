import Container from './Container';
import Description from './Description';
import ReadButton from './ReadButton';
import Title from './Title';

export default {
	Title,
	Description,
	Container,
	ReadButton,
};
