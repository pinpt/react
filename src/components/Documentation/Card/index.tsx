export interface IDocumentationCardProps {
	className?: string;
	title?: string | React.ReactElement;
	description?: string | React.ReactElement;
	cta?: string | React.ReactElement;
	onCtaClick?: () => void;
}

const Card = (props: IDocumentationCardProps) => {
	const {
		className = '',
		title,
		description,
		cta = 'Read',
		onCtaClick
	} = props;

	return (
		<div className={`Pinpoint Documentation Card Container wrapper ${className}`}>
			{title && (
				<h2 className="title">
					{title}
				</h2>
			)}
			{description && (
				<p className="description">
					{description}
				</p>
			)}
			{onCtaClick && (
				<div className="cta clickable" onClick={onCtaClick}>
					{cta}
				</div>
			)}
		</div>
	);
}

export default Card;