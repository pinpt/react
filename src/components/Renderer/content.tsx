import mediumZoom from 'medium-zoom';
import { forwardRef, useEffect } from 'react';
import { slugifyContent } from '../../lib/string';
import { CoverMediaType } from '../../lib/types/content';
import { Document } from './';
import React from 'react';
import type { ICoverMedia } from '../../lib/types/content';

const ImageMedia = ({ src, title = '', zoomable = false }: { src: string; title?: string; zoomable?: boolean }) => {
	useEffect(() => {
		if (typeof window !== 'undefined') {
			mediumZoom('.medium-zoom-cover');
		}
	}, []);

	return (
		<div className="Pinpoint image">
			<img src={src} alt={title} className={`${zoomable ? 'medium-zoom-cover' : ''}`} />
		</div>
	);
};

const VideoMedia = ({ src }: { src: string }) => {
	return (
		<div className="Pinpoint video">
			<video controls src={src} />
		</div>
	);
};

const YoutubeMedia = ({
	id,
	metadata,
}: {
	id: string;
	metadata?: Record<string, any>;
}) => {
	const posterUrl = `https://i.ytimg.com/vi/${id}/${metadata?.poster ?? 'hqdefault'}.jpg`;
	return (
		<div className="Pinpoint youtube">
			<div
				className="yt"
				style={{ backgroundImage: `url("${posterUrl}")` }}
				data-url={`https://www.youtube-nocookie.com/embed/${encodeURIComponent(id)}?autoplay=1`}
			>
				<button className="play-button" />
			</div>
		</div>
	);
};

export const CoverMedia = ({
	media,
	title,
	zoomable,
}: {
	media?: ICoverMedia;
	title?: string;
	zoomable?: boolean;
}) => {
	if (!media) {
		return <></>;
	}
	let content: React.ReactNode = null;
	switch (media.type) {
		case CoverMediaType.None: {
			return <></>;
		}
		case CoverMediaType.Image: {
			content = <ImageMedia src={media.value || media.placeholderImage} title={title} zoomable={zoomable} />;
			break;
		}
		case CoverMediaType.Video: {
			content = <VideoMedia src={media.value} />;
			break;
		}
		case CoverMediaType.Youtube: {
			content = <YoutubeMedia id={media.value} metadata={media.metadata} />;
			break;
		}
	}
	return (
		<section className="Pinpoint cover covermedia">
			<div className="Pinpoint media-container">{content}</div>
		</section>
	);
};

interface ContentProps {
	id: string;
	title?: string;
	document?: any;
	coverMedia?: ICoverMedia;
	limit?: number;
	divider?: boolean;
}

const Content = forwardRef((props: ContentProps, ref: any) => {
	return (
		<article ref={ref}>
			{props.coverMedia && <CoverMedia media={props.coverMedia} title={props.title} />}
			<section className="Pinpoint content">
				<Document node={props.document} limit={props.limit} />
				{props.limit && props.document.content?.length > props.limit && (
					<div className="Pinpoint continue">
						<a href={slugifyContent(props.id, props.title)}>Continue Reading â†’</a>
					</div>
				)}
			</section>
			{props.divider && <hr className="Pinpoint divider" />}
		</article>
	);
});

export default Content;
