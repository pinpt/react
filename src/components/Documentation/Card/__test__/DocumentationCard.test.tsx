import renderer from 'react-test-renderer';
import Card from '..';
import { IContent } from '../../../../lib/types/content';
import entries from '../../../Prebuilt/__data__/testEntries.json';
import site from '../../../Prebuilt/__data__/testSite.json';

test('Test default state', () => {
	const component = renderer.create(
		<Card
			title={entries[0].title}
			description={entries[0].headline}
			onCtaClick={() => console.log('action!')}
		/>
	);
	const tree = component.toJSON();
	expect(tree).toMatchSnapshot();
});