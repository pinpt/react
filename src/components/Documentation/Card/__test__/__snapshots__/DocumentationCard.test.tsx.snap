// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test default state 1`] = `
<div
  className="Pinpoint Documentation Card Container wrapper "
>
  <h2
    className="title"
  >
    Linked Content Blocks
  </h2>
  <p
    className="description"
  >
    Linked content blocks allow you to populate your content from the recently completed work. Annotations show you which issues and pull requests are linked.
  </p>
  <div
    className="cta clickable"
    onClick={[Function]}
  >
    Read
  </div>
</div>
`;
